### ISSUE TITLE
clang warnings when compiling uvloop on CPython 3.6

### ISSUE THREAD

**Issue by 1st1** [cython/cython#1513]
CPython version: 3.6
Cython version: 0.25.1

I have 100s of the following warnings when I compile uvloop with clang:

```
uvloop/loop.c:5995:7: warning: & has lower precedence than ==; == will be evaluated first [-Wparentheses]
  if (__Pyx_PyFastCFunction_Check(__pyx_t_7)) {
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
uvloop/loop.c:193:79: note: expanded from macro '__Pyx_PyFastCFunction_Check'
    ((PyCFunction_Check(func) && METH_FASTCALL == PyCFunction_GET_FLAGS(func) & ~(METH_CLASS | METH_STATIC | METH_COEXIST)))
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
uvloop/loop.c:5995:7: note: place parentheses around the '==' expression to silence this warning
  if (__Pyx_PyFastCFunction_Check(__pyx_t_7)) {
      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

and

```
uvloop/loop.c:29222:32: warning: explicitly assigning value of variable of type 'struct __pyx_obj_6uvloop_4loop___pyx_scope_struct_10_create_unix_connection *' to itself [-Wself-assign]
  if (1); else __pyx_cur_scope = __pyx_cur_scope;
               ~~~~~~~~~~~~~~~ ^ ~~~~~~~~~~~~~~~
uvloop/loop.c:29222:34: warning: code will never be executed [-Wunreachable-code]
  if (1); else __pyx_cur_scope = __pyx_cur_scope;
                                 ^~~~~~~~~~~~~~~
uvloop/loop.c:29222:7: note: silence by adding parentheses to mark code as explicitly dead
  if (1); else __pyx_cur_scope = __pyx_cur_scope;
      ^
      /* DISABLES CODE */ ( )
uvloop/loop.c:29222:7: warning: code will never be executed [-Wunreachable-code]
  if (1); else __pyx_cur_scope = __pyx_cur_scope;
      ^
```

**Comment by scoder** [comment]
Duplicate of #1506

**Comment by 1st1** [comment]
@scoder I don't think it's a complete duplicate. Please take a look at the second warning.

**Comment by scoder** [comment]
The second set of warnings is also fixed for Cython 0.26 in 6b1193e95845330f88190b15925d369961a1318b.
